bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -162,4 +161,24 @@ private String parseIdentifier(IdentifierLocation identifierLocation, SlingHttpS
             throw new RuntimeException("Identifier location " + identifierLocation + " is not supported");
         }
     }
+
+    static class StringSubstitutor {
```

Comment:
Instead of inventing our own new class can't we just use https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#replaceEach-java.lang.String-java.lang.String:A-java.lang.String:A- ? This bundle is already available in AEM.

Or as alternative use a regex?

Reply:
Our `StringSubstitutor:replace` method uses just that (`StringUtils.replaceEach`). Our own `StringSubstituor` was created to preserve the `UrlProviderImpl` implementation.

Reply:
I saw we use StringUtils.replaceEach, wonder if that also can be done simpler? Mabe with String.formart() or RegEx. But I'm just nitpicking here.